core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -1,123 +1,1058 @@
-  def fold[C](fa: A => C, fb: B => C)(implicit F: Functor[F]): F[C] = F.map(value)(_.fold(fa, fb))
+  /**
+   * Folds the underlying `Either` value, yielding an `F[_]` as result.
+   *
+   * Example:
+   * {{{
+   *   val parsed: Eval[Long] = parseNum(num).fold(_ => 0, x => x)
+   * }}}
+   */
+  def fold[C](fa: A => C, fb: B => C)(implicit F: Functor[F]): F[C] =
+    F.map(value)(_.fold(fa, fb))
 
-  def isLeft(implicit F: Functor[F]): F[Boolean] = F.map(value)(_.isLeft)
+  /**
+   * Queries the underlying `Either` value, returning `true` if it's a `Left`,
+   * or `false` otherwise.
+   *
+   * {{{
+   *   // Yields `false` on evaluation
+   *   EitherT.rightT[Eval, Throwable](10).isLeft
+   *
+   *   // Yields `true` on evaluation
+   *   EitherT.leftT[Eval, Int]("My error").isLeft
+   * }}}
+   *
+   * @see [[isRight]]
+   */
+  def isLeft(implicit F: Functor[F]): F[Boolean] =
+    F.map(value)(_.isLeft)
 
-  def isRight(implicit F: Functor[F]): F[Boolean] = F.map(value)(_.isRight)
+  /**
+   * Queries the underlying `Either` value, returning `true` if it's a `Right`,
+   * or `false` otherwise.
+   *
+   * {{{
+   *   // Yields `true` on evaluation
+   *   EitherT.rightT[Eval, String](10).isRight
+   *
+   *   // Yields `false` on evaluation
+   *   EitherT.leftT[Eval, Int]("My error").isRight
+   * }}}
+   *
+   * @see [[isLeft]]
+   */
+  def isRight(implicit F: Functor[F]): F[Boolean] =
+    F.map(value)(_.isRight)
 
-  def swap(implicit F: Functor[F]): EitherT[F, B, A] = EitherT(F.map(value)(_.swap))
+  /**
+   * Swaps the left and right values on evaluation, such that if the
+   * underlying `Either` is a `Left`, then returns a `Right` on the evaluation
+   * of `F[_]` or vice versa.
+   *
+   * {{{
+   *   val rt = EitherT.rightT[Eval, String](10)
+   *
+   *   rt.value.value
+   *   //=> res: Either[String,Int] = Right(10)
+   *
+   *   rt.swap.value.value
+   *   //=> res: Either[Int,String] = Left(10)
+   * }}}
+   */
+  def swap(implicit F: Functor[F]): EitherT[F, B, A] =
+    EitherT(F.map(value)(_.swap))
 
-  def getOrElse[BB >: B](default: => BB)(implicit F: Functor[F]): F[BB] = F.map(value)(_.getOrElse(default))
+  /**
+   * Casts the left type parameter to a super type.
+   *
+   * Like many types described in Cats, the type parameters of `EitherT`
+   * are invariant, even if the `Either` data type is covariant in both
+   * type parameters.
+   *
+   * This operator is a shortcut for safely upcasting the left type parameter
+   * to a compatible super type, an operation that would be automatic under
+   * Scala's type system if the parameters were declared as being covariant.
+   *
+   * Example:
+   *
+   * {{{
+   *   val num: EitherT[Eval, NumberFormatException, Long] =
+   *     EitherT.rightT(10210)
+   *
+   *   val num2: EitherT[Eval, Throwable, Long] =
+   *     num.upcastL
+   * }}}
+   *
+   * @see [[upcastR]]
+   */
+  @inline def upcastL[AA >: A]: EitherT[F, AA, B] =
```

Comment:
can we call this leftWiden to mirror widen which we normally use here?
https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Bifunctor.scala#L54